shared/aux.go

```diff
@@ -123,6 +123,16 @@ func PrintBase64Encoded(path string) error {
 	return nil
 }
 
+// AppendIfMissing appends a value to a slice if that value does not already exist in the slice.
+func AppendIfMissing(slice []Node, i Node) []Node {
```

Comment:
if your slice is type of []Node , lets give a more meaningful name to it ... maybe nodes ? idk

Reply:
Ah this is a great point... and I think also touches on this comment too:
> do u intend to export this func outside the pkg ?

because initially I was thinking "hey we can use this for any slice we want to append to!" forgetting entirely that the func signature is for Node. Soooo yeah I'll change the name and remove the export actually in this case!